### DISCUSSION TITLE
Support for removing the assigned jobs

### DISCUSSION THREAD

**Discussion by oneforalone** [filecoin-project/lotus#8055] [Storage Provider] [answered]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated improvement to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Improvement Suggestion

In some reason, i got overload jobs assigned for workers, i.e. got 9k jobs while workers can only run 51 jobs. I tried to auto remove the jobs and sectors by simple scripts, but it's such a pain cause i remove 2k jobs using 10 hours. So here i am asking team could tell me how to remove the assigned jobs, cause it can speed up my progress to removing the useless sealing jobs. Or is there any solutions for my problem, all i need is to removing all the old overload sealing jobs. Please help!!!

**Comment by Reiers** [comment, marked as answer]
**Hi @oneforalone** 

First - I think you need to write a better pledge script or slow down the deals - because 9k jobs is way too much, for a data cluster of machine that can only handle 51 jobs !
 
Second, this is isn't a lotus issue - the issue is created by the user so I will transfer the ticket over to discussions for help and troubleshooting. 

I do have environment if all things fail, shut down miner:
`export LOTUS_MINER_ABORT_UNFINISHED_WORK=1`
and `lotus-miner run`.

**Only use as last resort! - and keep in mind that this will most likely break other sectors if the workers is still working on other sectors.**

**Reply by oneforalone** [reply]
Thanks for your advices, i'll try a better pledge way.

However, I tried another way to clean it up before your reply. Here is my trick:
First, I create the empty sector files in worker's sealed directory. Then restart the worker, this will make the PC1 jobs failed.
And then I just remove the failed sectors. 

But i'm sure the last resort is what i want, really appreciate your reply.

**Comment by LexLuthr** [comment]
I am archiving this discussion as there has been no new updates from the author.
Please refer to https://github.com/filecoin-project/lotus/discussions/7228 for the discussion guidelines.
